### PR TITLE
Document writing portable config with storage.

### PR DESCRIPTION
**Pull request by erictune**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1311)

### CODE REVIEW

**Comment by erictune** [comment]
@viglesiasce you want this

@thockin we just talked about this an hour ago

@jsafrane @saad-ali we have talked about this issue before.

File: docs/user-guide/persistent-volumes/index.md [hunk 1]
```diff
@@ -399,3 +399,27 @@ parameters:
 
 * `type`: [VolumeType](http://docs.openstack.org/admin-guide/dashboard-manage-volumes.html) created in Cinder. Default is empty.
 * `availability`: Availability Zone. Default is empty.
+
+## Writing Portable Configuration
+
+When writing configuration templates or examples that need persistent storage
```
Comment (devin-donnelly) [inline docs/user-guide/persistent-volumes/index.md]: A little clearer if you reverse the order in this sentence.

"If you're writing configuration templates or examples that run on a wide range of clusters and need persistent storage, we recommend that you use the following pattern:"

File: docs/user-guide/persistent-volumes/index.md [hunk 2]
```diff
@@ -399,3 +399,27 @@ parameters:
...
+and which are intended to run on a wide range of clusters, we recommend the following pattern:
+
+- Do include PersistentVolumeClaim objects in your bundle of config (alongside Deployments, ConfigMaps, etc).
+- Do not include PersistentVolume objects in the config, since the user instantiating the config may not have
+  permission to create PersistentVolumes.
+- Give the user the option of providing a storage class name when instantating the template.
+- If the user provides a storage class name, and the cluster is version 1.4 or newer, put that value into the `volume.beta.kubernetes.io/storage-class` annotation of the PVC.
```
Comment (devin-donnelly) [inline docs/user-guide/persistent-volumes/index.md]: These next two bullets, related to storage class names, should be sub-bullets of the previous one (line 411).

Comment (erictune) [inline docs/user-guide/persistent-volumes/index.md]: done

File: docs/user-guide/persistent-volumes/index.md (hunk 1 above)
Comment (erictune) [inline docs/user-guide/persistent-volumes/index.md]: done

**Comment by thockin** [comment]
tech lgtm

**Comment by erictune** [comment]
I addressed the doc reviewers comments so marking this Doc LGTM.

**Comment by erictune** [comment]
Github Reviews doesn't seem to understand that I have addressed the review comments.

**Pull request merged**